firefox-ios/Client/Frontend/Home/TopSites/DataManagement/UnifiedAds/UnifiedAdsProvider.swift

```diff
@@ -6,21 +6,38 @@ import Common
 import Foundation
 import Shared
 
+typealias UnifiedTileResult = Swift.Result<[UnifiedTile], Error>
+
 /// Used only for sponsored tiles content and telemetry. This is aiming to be a temporary API
 /// as we'll migrate to using A-S for this at some point in 2025
 protocol UnifiedAdsProviderInterface {
-    func fetchTiles(completion: @escaping (ContileResult) -> Void)
+    /// Fetch contiles either from cache or backend
+    /// - Parameters:
+    ///   - timestamp: The timestamp to retrieve from cache, useful for tests. Default is Date.now()
+    ///   - completion: Returns an array of Tiles, can be empty
+    func fetchTiles(timestamp: Timestamp, completion: @escaping (UnifiedTileResult) -> Void)
+}
+
+extension UnifiedAdsProviderInterface {
+    func fetchTiles(timestamp: Timestamp = Date.now(), completion: @escaping (UnifiedTileResult) -> Void) {
+        fetchTiles(timestamp: timestamp, completion: completion)
+    }
 }
 
-class UnifiedAdsProvider: UnifiedAdsProviderInterface {
-    private static let resourceEndpoint = "https://ads.mozilla.org/v1/ads"
+class UnifiedAdsProvider: URLCaching, UnifiedAdsProviderInterface, FeatureFlaggable {
+    private static let prodResourceEndpoint = "https://ads.mozilla.org/v1/ads"
+    static let stagingResourceEndpoint = "https://ads.allizom.org/v1/ads"
```

Comment:
are we sure we want the ads urls public? Has this been discussed?

Reply:
Mhh, so this one is public as the one is public in `ContileProvider`. It's public to be accessed by the unit tests. It's not a good reasons to make it public, so happy to change this. I guess I could duplicate it for unit tests if it's not public?

Reply:
Just asking that we're ok with that. For wallpapers, for example, it was explicitly not made public, so I'm just checking.